app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PagesController < ApplicationController
+  before_action :find_page
+
+  def show
+    render @page
+  end
+
+  private
+
+  def find_page
+    id = params.require(:id)
```

Comment:
`params.permit(:id).require(:id)` ?

Reply:
Good catch!

Reply:
Why permit is needed? Code around doesn't care about permitting at all.

Reply:
it ensures the `id` is a scalar

Reply:
This is the only way to avoid allowing external users to trigger 500s by sending `?id[]=fake` or `?id[key]=fake`.

Always permit before you require. Look across rubygems.org code for more examples.

Reply:
I understand, but what is the difference in here? Array or Hash will not break the only usage of `include?` later. It will generate proper 404 response since not found in PAGES constant. Happy to change this if still needed to keep it "consistent".

Reply:
I think the bugs come later, when years from now someone assumes it's a scalar and does something other than `include?`.